edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet.Docker/SystemResourcesMetrics.cs

```diff
@@ -195,82 +205,89 @@ void SetModuleStats(SystemResources systemResources)
             }
         }
 
-        Option<double> GetCpuUsage(DockerStats module, string name)
+        Option<double> GetCpuUsage(DockerStats module)
         {
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
             {
-                // Get values if exist
-                double totalUsage = 0, systemUsage = 0;
-                if (!module.CpuStats.Exists(cpuStats => cpuStats.CpuUsage.Exists(cpuUsage => cpuUsage.TotalUsage.Exists(tu =>
-                     {
-                         totalUsage = tu;
-                         return true;
-                     })) && cpuStats.SystemCpuUsage.Exists(su =>
-                     {
-                         systemUsage = su;
-                         return true;
-                     })))
-                {
-                    // One of the values is missing, skip.
-                    return Option.None<double>();
-                }
+                return this.GetCpuLinux(module);
+            }
+            else
+            {
+                return this.GetCpuWindows(module);
+            }
+        }
+
+        // Modeled after https://github.com/docker/cli/blob/master/cli/command/container/stats_helpers.go#L166
```

Comment:
This comment will go stale when docker cli code changes. Refer to a specific commit? If you want to keep the ugliness factor down, the URL only needs 7-10 digits of the commit ID...